Code-the-Dream-School/dd-prac-team1-front
# Dev/shopping list (Part 1) (PRA-134)

## One Line Description
putting recipe in shopping list and rendering recipe
## Requirements

_What this code should do_
This code allow you to add the recipe to the shopping list (the modal window that allows you to change the serving size is commented right now till the new backend updates). After the recipe added you will see the confirming in the console. After you will put all recipes you need to shopping list go to the shopping list. You should see the list of ingredients and page layout. You also could add new ingredient but you cannot save it yet and you will also could mention the warning on the console log because this new ingredient doesn't have an ID yet.
## Notes
Because Backend and Frontend are working on this functional on real time not all functional available yet. Next pull request we will introduce more functional and logic. Right now we mixed up with counting amounts and units, we will fix it vary soon.
_Any additional info_

## Test Steps
go to saved recipe, choose recipe and click on it. In the single recipe click on the shopping list icon 
<img width="44" alt="image" src="https://github.com/Code-the-Dream-School/dd-prac-team1-front/assets/106414157/526a612c-68a6-45ac-a1e1-71989f385663">

after go to the shopping list using navbar. You'll see list of ingredients. Check that you are able to add new ingredient (you can't save it yet)
_Instructions on how to test the changes_

## UI/UX Outcome
button to save recipe to the shopping list: 
<img width="90" alt="image" src="https://github.com/Code-the-Dream-School/dd-prac-team1-front/assets/106414157/69102ce1-3869-4c9a-add5-2b6a1018cda3">

shopping list in Navbar 
<img width="202" alt="image" src="https://github.com/Code-the-Dream-School/dd-prac-team1-front/assets/106414157/b41e879f-ec5d-4d79-8bfe-3d6ff05ce9a5">

Try to add new ingredient:
<img width="713" alt="image" src="https://github.com/Code-the-Dream-School/dd-prac-team1-front/assets/106414157/e8a98082-a576-47a4-8dcb-61d21ea6b354">


Shopping list
<img width="878" alt="image" src="https://github.com/Code-the-Dream-School/dd-prac-team1-front/assets/106414157/a6cceb21-032a-4e7d-8e9d-3311967b0b2d">




_Attach a screenshot here_

## Checklist

- The Linear ID is in the PR title
- My code follows best practices
- Documentation is included where needed
